### ISSUE TITLE
Unhandled Error: KeyError: 139255 during POS tagging using the FR/ES model

### ISSUE THREAD

**Issue by pocman** [RasaHQ/rasa_nlu#580]
**rasa NLU version** (e.g. `0.7.3`): 0.10.0a3

**Used backend / pipeline** (`mitie`, `spacy_sklearn`, ...): Dockerfile_full

**Operating system** (windows, osx, ...): docker & linux

**Issue**:
Here is the error on spacy : https://github.com/explosion/spaCy/issues/1052

Running the following query `/parse?q=t%20+3` will raise a 500 while `/parse?q=t+3` won't.
It could be nice to handle this error and avoid 500.

Here is the stack trace :
```
2017-09-11 12:34:39+0000 [-] Unhandled Error
        Traceback (most recent call last):
          File "/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 500, in errback
            self._startRunCallbacks(fail)
          File "/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 567, in _startRunCallbacks
            self._runCallbacks()
          File "/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 653, in _runCallbacks
            current.result = callback(current.result, *args, **kw)
          File "/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1442, in gotResult
            _inlineCallbacks(r, g, deferred)
        --- <exception caught here> ---
          File "/usr/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1384, in _inlineCallbacks
            result = result.throwExceptionIntoGenerator(g)
          File "/usr/local/lib/python2.7/site-packages/twisted/python/failure.py", line 393, in throwExceptionIntoGenerator
            return g.throw(self.type, self.value, self.tb)
          File "/app/rasa_nlu/server.py", line 127, in parse_get
            else threads.deferToThread(self.data_router.parse, data))
          File "/usr/local/lib/python2.7/site-packages/twisted/python/threadpool.py", line 250, in inContext
            result = inContext.theWork()
          File "/usr/local/lib/python2.7/site-packages/twisted/python/threadpool.py", line 266, in <lambda>
            inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
          File "/usr/local/lib/python2.7/site-packages/twisted/python/context.py", line 122, in callWithContext
            return self.currentContext().callWithContext(ctx, func, *args, **kw)
          File "/usr/local/lib/python2.7/site-packages/twisted/python/context.py", line 85, in callWithContext
            return func(*args,**kw)
          File "rasa_nlu/data_router.py", line 200, in parse
            response = model.parse(data['text'], data.get('time', None))
          File "rasa_nlu/model.py", line 263, in parse
            component.process(message, **self.context)
          File "rasa_nlu/utils/spacy_utils.py", line 80, in process
            message.set("spacy_doc", self.nlp(message.text))
          File "/usr/local/lib/python2.7/site-packages/spacy/language.py", line 329, in __call__
            proc(doc)
          File "spacy/tagger.pyx", line 212, in spacy.tagger.Tagger.__call__ (spacy/tagger.cpp:6245)
            self.vocab.morphology.assign_tag_id(&tokens.c[i], guess)
          File "spacy/morphology.pyx", line 76, in spacy.morphology.Morphology.assign_tag_id (spacy/morphology.cpp:4681)
            tag_id = self.reverse_index[self.strings['SP']]
        exceptions.KeyError: 139255
```

**Content of configuration file** (if used & relevant):
```json
{
    "entity_synonyms": null, 
    "pipeline": [
        "nlp_spacy", 
        "tokenizer_spacy", 
        "intent_featurizer_spacy", 
        "ner_crf", 
        "ner_synonyms", 
        "intent_classifier_sklearn", 
        "ner_duckling"
    ], 
    "entity_extractor_crf": null, 
    "language": "en", 
    "intent_classifier_sklearn": "intent_classifier.pkl", 
    "trained_at": "20170908-160809", 
    "training_data": "training_data.json", 
    "spacy_model_name": "en", 
    "rasa_nlu_version": "0.10.0a3", 
    "ner_duckling_persisted": "ner_duckling.json"
}
```

```json
{
    "aws_endpoint_url": null,
    "response_log": "/app/logs",
    "max_number_of_ngrams": 7,
    "ner_crf": {
        "BILOU_flag": true,
        "L2_c": 0.001,
        "L1_c": 1,
        "features": [
            [
                "low",
                "title",
                "upper",
                "pos",
                "pos2"
            ],
            [
                "bias",
                "low",
                "word3",
                "word2",
                "upper",
                "title",
                "digit",
                "pos",
                "pos2",
                "pattern"
            ],
            [
                "low",
                "title",
                "upper",
                "pos",
                "pos2"
            ]
        ],
        "max_iterations": 50
    },
    "cors_origins": [],
    "port": 5000,
    "log_level": "INFO",
    "mitie_file": "data/total_word_feature_extractor.dat",
    "duckling_dimensions": null,
    "server_model_dirs": null,
    "spacy_model_name": null,
    "config": "config.json",
    "emulate": null,
    "nlu_python_packages": "/usr/local/lib/python2.7/dist-packages",
    "path": "/app/./models",
    "data": "./data/examples/rasa/demo-rasa.json",
    "num_threads": 1,
    "pipeline": [
        "nlp_spacy",
        "tokenizer_spacy",
        "intent_featurizer_spacy",
        "ner_crf",
        "ner_synonyms",
        "intent_classifier_sklearn",
        "ner_duckling"
    ],
    "name": null,
    "language": "en",
    "intent_classifier_sklearn": {
        "kernel": "linear",
        "C": [
            1,
            2,
            5,
            10,
            20,
            100
        ]
    },
    "max_training_processes": 1,
    "token": null,
    "nlu_home": "/app",
    "nlu_docker": "YES",
    "log_file": null
}
```

```
## intent:affirm
- yes
- yep
- yeah
- indeed
- that's right
- ok
- great
- right, thank you
- correct
- great choice
- sounds really good

## intent:goodbye
- bye
- goodbye
- good bye
- stop
- end
- farewell
- Bye bye
- have a good one

## intent:greet
- hey
- howdy
- hey there
- hello
- hi
- good morning
- good evening
- dear sir

## intent:restaurant_search
- i'm looking for a place to eat
- I want to grab lunch
- I am searching for a dinner spot
- i'm looking for a place in the [north](location) of town
- show me [chinese](cuisine) restaurants
- show me [chines](cuisine:chinese) restaurants
- show me a [mexican](cuisine) place in the [centre](location)
- i am looking for an [indian](cuisine) spot called olaolaolaolaolaola
- search for restaurants
- anywhere in the [west](location)
- anywhere near [18328](location)
- I am looking for [asian fusion](cuisine) food
- I am looking a restaurant in [29432](location)
- I am looking for [mexican indian fusion](cuisine)
- [central](location) [indian](cuisine) restaurant

## synonym:chinese
- Chines
- chines
- Chinese

## synonym:vegetarian
- vegg
- veggie
```

**Comment by tmbo** [comment]
I think it is a good idea to check with the spacy people - I am not sure what is going on there. (maybe the space is missing from the internal dictionary).

**Comment by pocman** [comment]
Yes, it's a known issue with Spacy. 
I linked this issue to this one, I hope it will add some weight on their side :/ https://github.com/explosion/spaCy/issues/1052#issuecomment-328517854

**Comment by tmbo** [comment]
Did you train your own model? If so it seems there is a fix in spacy 1.9.0 that adds the missing symbol to the model.

**Comment by tmbo** [comment]
Should be fixed in spacy 1.9.0 I guess. let me know if there are still issues with this.
